evap/evaluation/views.py

```diff
@@ -75,6 +75,12 @@ def index(request):
             # clean up our test cookie
             if request.session.test_cookie_worked():
                 request.session.delete_test_cookie()
+
+            # check for redirect variable
+            redirect_to = request.GET.get("next", None)
+            if redirect_to is not None:
+                return redirect(redirect_to)
+
```

Comment:
Why did this not work before? Redirecting to "evaluation:index" means the browser would send a GET request for this view, again. Then, the next cookie should still be set, and the "correct" redirect should be handled in lines 100 to 106.

There is already a test for redirecting to the correct start page -- see evaluation/tests/test_views.py line 25 (`test_login_for_staff_users_correctly_redirects`). If I read it correctly, this asserts the redirection to `evaluation:index` followed by the redirection to the correct page. This makes me think there is something wrong with this snippet of code.

In general, I think we just want this branch to fall through to the end of the view, where the case that a user logged in is handled, skipping the whole double-redirect-thing.

Reply:
Ah ok, so I missed that we're not talking about a cookie here, but just a GET parameter. Then, I see that it is lost on the redirect.

I still think that we should simply do nothing here in this if-block and just let the program flow continue until it reaches the handling for the next parameter below in lines 100 to 106.